helm/cluster/templates/clusterapi/_helpers.tpl

```diff
@@ -0,0 +1,36 @@
+{{- define "cluster.internal.kubeadm.proxy.noProxyList" }}
```

Comment:
Helper that renders `NO_PROXY` env var. It first adds predefined values:
- local addresses
- cluster domain
- services CIDRs
- pods CIDRs

And then also adds custom addresses:
- custom addresses
  - cluster-$provider apps can specify provider-specific addresses
- name of Helm template that renders a list of addresses
  - cluster-$provider apps can specify their own provider-specific templates that render addresses.
  - This can be used in advanced scenarios where some address should be calculated so it's not possible to pass a static value in Helm values.

The logic above is mostly taken from `cluster-aws` and then made configurable so we can use it also for other providers.